backend/tests/api-e2e/test_api_deposits.py

```diff
@@ -0,0 +1,167 @@
+import pytest
+
+from app.extensions import w3
+
+from tests.conftest import Client, UserAccount
+from tests.helpers.constants import STARTING_EPOCH
+from flask import current_app as app
+
+# Please note that tests here assume that they talk to blockchain and indexer
+# whose state is not reset between tests.
+
+
+@pytest.mark.api
+def test_deposit_basics(
+    client: Client,
+    deployer: UserAccount,
+    ua_alice: UserAccount,
+    ua_bob: UserAccount,
+    setup_funds,
+):
+    # Check effective deposit before first GLM lock
+    (
+        total_effective_estimated_before_lock,
+        response_code,
+    ) = client.get_total_effective_estimated()
+    assert int(total_effective_estimated_before_lock["totalEffective"]) == 0
+    assert response_code == 200
+
+    user_deposit1_before_lock, response_code = client.get_user_deposit(
+        ua_alice.address, STARTING_EPOCH
+    )
+    assert int(user_deposit1_before_lock["effectiveDeposit"]) == 0
+    assert response_code == 200
+
+    (
+        user_estimated_deposit_before_lock,
+        response_code,
+    ) = client.get_user_estimated_effective_deposit(ua_alice.address)
+    assert int(user_estimated_deposit_before_lock["effectiveDeposit"]) == 0
+    assert response_code == 200
+
+    # Lock GML for one account
+    alice_GLM_budget = 10000
+    alice_first_lock = 8000
+    deployer.transfer(ua_alice, alice_GLM_budget)
+    ua_alice.lock(alice_first_lock)
+
+    indexed_height = client.wait_for_height_sync()
+    app.logger.debug(f"indexed blockchain height: {indexed_height}")
+
+    # Check effective deposit after first GLM lock
+    user_deposit_after_lock1, _ = client.get_user_deposit(
+        ua_alice.address, STARTING_EPOCH
+    )
+    assert (
+        0
+        < int(user_deposit_after_lock1["effectiveDeposit"])
+        < w3.to_wei(alice_first_lock, "ether")
+    )
```

Comment:
This doesn't work. Order of evaluation is like this:
```
0 < 1 -> True
True < 2 -> True
```
Where second step doesn't make any sense.

Reply:
You can replicate this behavior by placing parenthesis around arguments:

```
(0 < 1) < 2
```

Reply:
Hmm, according to Python docs, this construction is allowed and has special handling. 
`a < b < c` expands to `a < b and b < c`. Anyway, please don't do that since when such assertion fails, it is not clear if left side or right side is to blame.

Reply:
https://docs.python.org/3/reference/expressions.html#comparisons